volestipy/volestipy/src/bindings.cpp

```diff
@@ -119,14 +116,77 @@ double HPolytopeCPP::generate_samples(int walk_len, int number_of_points, int nu
             }
         }
    }
-   
-   std::cout<<"sampling completed"<<std::endl;
+   std::cout<<"Sampling completed"<<std::endl;
 
-// The following block of code should NOT be removed!
+// The following block of code allows us to parse the matrix with the points we are making
    auto n_si=0;
    for (auto it_s = rand_points.begin(); it_s != rand_points.end(); it_s++){
       for (auto i = 0; i != it_s->dimension(); i++){
          samples[n_si++] = (*it_s)[i];
       }
    }
 }
+//////////         end of "generate_samples()"          //////////
+
+
+//////////         start of "rounding()"          //////////
+
+void HPolytopeCPP::rounding(char* rounding_method, double* new_A, double* new_b, double* T_matrix, double* shift, double &round_value){
+
+   // make a copy of the initial HP which will be used for the rounding step
+   auto P(HP);
+   RNGType rng(P.dimension());
+   CheBall = P.ComputeInnerBall();
+   
+   // set the output variable of the rounding step
+   round_result round_res;
+   
+   // walk length will always be equal to 2
+   int walk_len = 2;
```

Comment:
why not set by the user?

Reply:
We talked about this with Tolis and we said there is no need to.

Reply:
I think it is better to set it as a default choice since there is not any improvement inscreasing the walk length

Reply:
ok thanks for the explanation, it could be useful to state this in a comment